src/Core/Echo.Ast/Construction/AstParserContext.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Collections.Generic;
+using Echo.ControlFlow;
+using Echo.Core.Code;
+using Echo.DataFlow;
+
+namespace Echo.Ast.Construction
+{
+    internal sealed class AstParserContext<TInstruction>
+    {
+        private readonly ControlFlowGraph<TInstruction> _controlFlowGraph;
+
+        internal AstParserContext(
+            ControlFlowGraph<TInstruction> controlFlowGraph,
+            DataFlowGraph<TInstruction> dataFlowGraph)
+        {
+            _controlFlowGraph = controlFlowGraph;
+            DataFlowGraph = dataFlowGraph;
+        }
+
+        internal IInstructionSetArchitecture<TInstruction> Architecture => _controlFlowGraph.Architecture;
+
+        internal Dictionary<TInstruction, AstVariable[]> StackSlots
+        {
+            get;
+        } = new Dictionary<TInstruction, AstVariable[]>();
+
+        internal Dictionary<ExternalDataSourceNode<TInstruction>, AstVariable> ExternalSources
+        {
+            get;
+        } = new Dictionary<ExternalDataSourceNode<TInstruction>, AstVariable>();
+
+        internal Dictionary<IVariable, int> VariableVersions
+        {
+            get;
+        } = new Dictionary<IVariable, int>();
+
+        internal Dictionary<VariableSnapshot, AstVariable> VersionedVariables
+        {
+            get;
+        } = new Dictionary<VariableSnapshot, AstVariable>();
+
+        internal Dictionary<List<AstVariable>, AstVariable> VariableSourcesToPhiVariable
+        {
+            get;
+        } = new Dictionary<List<AstVariable>, AstVariable>(new AstVariableListComparer());
+
+        internal Dictionary<TInstruction, Dictionary<IVariable, int>> VariableStates
+        {
+            get;
+        } = new Dictionary<TInstruction, Dictionary<IVariable, int>>();
+
+        internal DataFlowGraph<TInstruction> DataFlowGraph
+        {
+            get;
+        }
+
+        internal int PhiCount
+        {
+            get;
+            set;
```

Comment:
Unnecessary public setter.

Reply:
PhiCount is incremented [here](https://github.com/Washi1337/Echo/pull/72/files/4e7f9ab5b72c059c289ca5b2c338c489b6caac6e#diff-3de9e1fb97e5976465b7d36d1d41d25f642126ad5e168ac5b43caf3d5d6540a7R293)